src/thirdparty/gpacmp4/mp4.c

```diff
@@ -528,189 +527,255 @@ int processmp4 (struct lib_ccx_ctx *ctx, struct ccx_s_mp4Cfg *cfg, char *file)
 			cc_track_count++;
 		if (type == GF_ISOM_MEDIA_VISUAL && subtype == GF_ISOM_SUBTYPE_AVC_H264)
 			avc_track_count++;
 	}
 
 	mprint("MP4: found %u tracks: %u avc and %u cc\n", track_count, avc_track_count, cc_track_count);
 
-	for (i = 0; i < track_count; i++) {
+	for(i = 0; i < track_count; i++)
+	{
 		const u32 type = gf_isom_get_media_type(f, i + 1);
 		const u32 subtype = gf_isom_get_media_subtype(f, i + 1, 1);
-		mprint("Processing track %d, type=%c%c%c%c subtype=%c%c%c%c\n", i + 1,
-			(unsigned char)(type >> 24 % 0x100), (unsigned char)((type >> 16) % 0x100),
-			(unsigned char)((type >> 8) % 0x100), (unsigned char)(type % 0x100),
-			(unsigned char)(subtype >> 24 % 0x100), (unsigned char)((subtype >> 16) % 0x100),
-			(unsigned char)((subtype >> 8) % 0x100), (unsigned char)(subtype % 0x100));
-
-		const u64 track_type = MEDIA_TYPE(type, subtype);
 
-		switch (track_type) {
-		case MEDIA_TYPE(GF_ISOM_MEDIA_VISUAL, GF_ISOM_SUBTYPE_XDVB): //vide:xdvb
+		if ( type == GF_ISOM_MEDIA_VISUAL && subtype == GF_ISOM_SUBTYPE_XDVB)
+		{
 			if (cc_track_count && !cfg->mp4vidtrack)
 				continue;
-			// If there are multiple tracks, change fd for different tracks
-			if (avc_track_count > 1) {
-				switch_output_file(ctx, enc_ctx, i);
-			}
-			if (process_xdvb_track(ctx, file, f, i + 1, &dec_sub) != 0) {
-				mprint("Error on process_xdvb_track()\n");
+			if(process_xdvb_track(ctx, file, f, i + 1, &dec_sub) != 0)
+			{
+				mprint("error\n");
 				free(dec_ctx->xds_ctx);
 				return -3;
 			}
-			if (dec_sub.got_output) {
+			if(dec_sub.got_output)
+			{
 				mp4_ret = 1;
 				encode_sub(enc_ctx, &dec_sub);
 				dec_sub.got_output = 0;
 			}
-			break;
+		}
 
-		case MEDIA_TYPE(GF_ISOM_MEDIA_VISUAL, GF_ISOM_SUBTYPE_AVC_H264): // vide:avc1
+		if( type == GF_ISOM_MEDIA_VISUAL && subtype == GF_ISOM_SUBTYPE_AVC_H264)
+		{
 			if (cc_track_count && !cfg->mp4vidtrack)
 				continue;
-			// If there are multiple tracks, change fd for different tracks
-			if (avc_track_count > 1) {
-				switch_output_file(ctx, enc_ctx, i);
-			}
-			GF_AVCConfig *cnf = gf_isom_avc_config_get(f, i + 1, 1);
-			if (cnf != NULL) {
-				for (j = 0; j < gf_list_count(cnf->sequenceParameterSets); j++) {
-					GF_AVCConfigSlot* seqcnf = (GF_AVCConfigSlot*)gf_list_get(cnf->sequenceParameterSets, j);
+			GF_AVCConfig *cnf = gf_isom_avc_config_get(f,i+1,1);
+			if (cnf!=NULL)
+			{
+				for (j=0; j<gf_list_count(cnf->sequenceParameterSets);j++)
+				{
+					GF_AVCConfigSlot* seqcnf=(GF_AVCConfigSlot* )gf_list_get(cnf->sequenceParameterSets,j);
 					do_NAL(enc_ctx, dec_ctx, (unsigned char *)seqcnf->data, seqcnf->size, &dec_sub);
+					//do_NAL (dec_ctx, (unsigned char *) seqcnf->data, seqcnf->size, &dec_sub);
 				}
 			}
-			if (process_avc_track(ctx, file, f, i + 1, &dec_sub) != 0) {
-				mprint("Error on process_avc_track()\n");
+
+			if(process_avc_track(ctx, file, f, i + 1, &dec_sub) != 0)
+			{
+				mprint("error\n");
```

Comment:
I pretty much prefer errors message to be specific. Error on process_avc_track() is much better than "error" (same everywhere else).

Reply:
Will keep in mind.